src/ast/semantic_analyser.cpp

```diff
@@ -1190,6 +1190,26 @@ void SemanticAnalyser::visit(Call &call)
     call.type = call.vargs->front()->type;
     call.type.SetAS(as);
   }
+  else if (call.func == "macaddr")
+  {
+    if (!check_nargs(call, 1))
+      return;
+
+    auto &arg = call.vargs->at(0);
+
+    if (!arg->type.IsIntTy() && !arg->type.IsArray() && !arg->type.IsPtrTy())
+      LOG(ERROR, call.loc, err_)
+          << "() only supports int, array, or pointer arguments"
```

Comment:
```suggestion
          << "() only supports array or pointer arguments"
```

Using an actual int as mac doesn't work as you're probe_reading them. Only ints that are actually pointers (like arg0 can be) would work.

Reply:
Ah yep, the int was from an earlier iteration and doesn't apply now